third_party/move/move-compiler/src/inlining/translate.rs

```diff
@@ -59,33 +90,47 @@ pub fn run_inlining(env: &mut CompilationEnv, prog: &mut Program) {
         visibilities: BTreeMap::new(),
         inline_stack: Default::default(),
         rename_counter: 0,
+        pre_compiled_lib,
+        processing_library: false,
     }
     .run(prog)
 }
 
 impl<'l> Inliner<'l> {
     fn run(&mut self, prog: &mut Program) {
         // First collect all definitions of inlined functions so we can expand them later in the AST.
-        self.visit_functions(prog, VisitingMode::All, &mut |ctx, fname, fdef| {
+        let mut find_inline_functions = |ctx: &mut Inliner, fname: &str, fdef: &Function| {
             if let Some(mid) = ctx.current_module {
                 let global_name = (mid, ctx.current_function);
                 ctx.visibilities
                     .insert(global_name, fdef.visibility.clone());
                 if fdef.inline {
                     if !matches!(fdef.body.value, FunctionBody_::Defined(_)) {
-                        ctx.env.add_diag(diag!(
-                            Inlining::Unsupported,
-                            (
-                                fdef.body.loc,
-                                format!("inline function {} must not be native", fname)
-                            )
-                        ));
+                        if !ctx.processing_library {
```

Comment:
For a pre-compiled library, inline functions may not have a body?

Reply:
It can be native, I think.